lib/child_process.js

```diff
@@ -855,7 +855,16 @@ function spawnSync(file, args, options) {
 
   // We may want to pass data in on any given fd, ensure it is a valid buffer
   for (let i = 0; i < options.stdio.length; i++) {
-    const input = options.stdio[i] && options.stdio[i].input;
+    if (!options.stdio[i]) {
+      continue;
+    }
+    const type = options.stdio[i].type;
+    if (type === 'overlapped') {
```

Comment:
should we throw only on windows?

Reply:
`overlapped` is not supported for child_process *Sync APIs(See [link](https://github.com/nodejs/node/blob/main/src/spawn_sync.cc#L896C24-L896C40)). I'm not sure if we can support this feature in the future.